ObjectiveGit/GTRepository.h

```diff
@@ -201,6 +211,18 @@ typedef NS_ENUM(NSInteger, GTRepositoryStateType) {
 /// Returns the initialized repository, or nil if an error occurred.
 + (nullable instancetype)repositoryWithURL:(NSURL *)localFileURL error:(NSError **)error;
 
+/// Convenience class initializer to find and open a repository with extended controls.
+///
+/// localFileURL - The file URL for the new repository. Cannot be nil.
+/// flags        - A combination of the `GTRepositoryOpenFlags` flags.
+/// ceilingDirs  - A GIT_PATH_LIST_SEPARATOR delimited list of path prefixes at
+///                which the search for a containing repository should terminate.
+///                Can be NULL.
+/// error        - The error if one occurs.
+///
+/// Returns the initialized repository, or nil if an error occurred.
++ (nullable instancetype)initWithURL:(NSURL *)localFileURL flags:(UInt32)flags ceilingDirs:(nullable const char *)ceilingDirs error:(NSError **)error;
```

Comment:
- Others class-methods for `GTRepository` are using `initWithURL:...`. It make sense for me to stay consistent. (Plus, I don't know how `repositoryWithURL...` will translate into Swift.  I will experiment and see)

- `NSArray <NSURL *> *` it is 😃